conda-env/dev.yml

```diff
@@ -12,15 +13,16 @@ dependencies:
     - cf_xarray=0.7.4
     - cftime=1.6.1
     - dask=2022.8.0
-    - esmpy=8.2.0
```

Comment:
Note, `esmpy` is never imported in the codebase so it isn't necessary to list as a package or environment dependency. It is implicitly installed with `xesmf`.

I removed `esmpy` from the conda envs and will remove it from the package recipe.

Just an FYI @jasonb5.